### DISCUSSION TITLE
Is this now the recommended way to set dependency version numbers? gradle/libs.versions.toml?

### DISCUSSION THREAD

**Discussion by nathanmeade** [android/nowinandroid#644] [Q&A] [answered]
It looks like to me that the version numbers for dependencies are set in this file. Is this now the recommended approach? Is this to guarantee that all of the modules are using the same versions if they are using the same dependency?

**Comment by PabloProta** [comment]
Yes, is a new approach to handle our dependencies. This is called Versions Catalogs, we use this "toml" file to set all dependencies, plugins and an extra feature called bundles, that allow us to inject a group of depencies at once instead inject one by one in the module that need them. 
you can check [here](https://docs.gradle.org/current/userguide/platforms.html) about version catalogs is a Gradle feature.
This project use a special module that is a [included build](https://docs.gradle.org/current/userguide/composite_builds.html), it's used to configure the version catalogs, you can check the "build-logic" that is a included build/composite build, its settings.gradle contains a source code that setup the version catalog.

**Comment by SimonMarquis** [comment, marked as answer]
@nathanmeade yes, you can check the official Android docs: 

> [Gradle version catalogs](https://docs.gradle.org/current/userguide/platforms.html) enable you to add and maintain dependencies and plugins in a scalable way. Using Gradle version catalogs makes managing dependencies and plugins easier when you have [multiple modules](https://developer.android.com/topic/modularization). Instead of hardcoding dependency names and versions in individual build files and updating each entry whenever you need to upgrade a dependency, you can create a central version catalog of dependencies that various modules can reference in a type-safe way with Android Studio assistance.
> **Source: https://developer.android.com/build/migrate-to-catalogs**

Also, don't forget to close this discussion and mark it as resolved if it's ok for you 😉
